integrations/extensions/docs/elasticsearch-install-and-setup/how_to_configure_advanced_elasticsearch_settings.md

```diff
@@ -0,0 +1,307 @@
+* `.elser_model_2_linux-x86-64` is the model ID for the optimized version of ELSER v2. It is recommended to use if it is available in your Elasticsearch deployment. Otherwise, use `.elser_model_2` for the regular ELSER v2 model, or `.elser_model_1` for ELSER v1.
+* `$QUERY` is the variable for accesing the user query. It will make sure that the user query will be passed to the query body.
+* `$FILTER` is the variable for accessing the customer filters configured either in the `Advanced Elasticsearch Settings` or when calling the search in an action step. It will make sure that the custom filters will be used in the query body.
+* Learn more about ELSER from [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-search-elser.html)
+* Learn more about Elasticsearch boolean query and filters from [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html)
+
+### KNN dense vector search
+```json
+{
+  "knn": {
+    "field": "text_embedding.predicted_value",
+    "query_vector_builder": {
+      "text_embedding": {
+        "model_id": "intfloat__multilingual-e5-small",
+        "model_text": "$QUERY"
+      }
+    },
+    "k": 10,
+    "num_candidates": 100,
+    "filter" : "$FILTER"
+  }
+}
+```
+Notes:
+* `text_embedding.predicted_value` refers to the field that stores the dense vectors. You may need to update it if you use a different field in your index.
+* `text_embedding` under `query_vector_builder` is the natural language processing task to perform. It has to be `text_embedding` for KNN search.
+* `intfloat__multilingual-e5-small` is the embedding model ID. You may need to update it if you want to use a different embedding model.
+* `$QUERY` is the variable for accesing the user query. It will make sure that the user query will be passed to the query body.
+* `$FILTER` is the variable for accessing the customer filters configured either in the `Advanced Elasticsearch Settings` or when calling the search in an action step. It will make sure that the custom filters will be used in the query body.
+* Learn more about knn search from [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html).
+* Learn more about how to set up a text embedding model in Elasticsearch from [here](text_embedding_deploy_and_use.md).
+
+### Using a nested query to search over nested documents with ELSER
+```json
+{
+  "query": {
+    "nested": {
+      "path": "passages",
+      "query": {
+        "bool": {
+          "should": [
+            {
+              "text_expansion": {
+                "passages.sparse.tokens": {
+                  "model_id": ".elser_model_2_linux-x86_64",
+                  "model_text": "$QUERY"
+                }
+              }
+            }
+          ],
+          "filter": "$FILTER"
+        }
+      },
+      "inner_hits": {"_source": {"excludes": ["passages.sparse"]}}
+    }
+  },
+  "_source": false
+}
```

Comment:
If I understand this correctly, this query body with the "filter" inside the "nested" block works only if you are trying to filter on properties of the inner hits.  I think if you want to filter on outer hit properties you need something like this:

```
{
    "query": {
        "bool": {
            "must": [
                {
                    "nested": {
                        "path": "passages",
                        "query": {
                            "bool": {
                                "should": [
                                    {
                                        "text_expansion": {
                                            "passages.sparse.tokens": {
                                                "model_id": ".elser_model_2_linux-x86_64",
                                                "model_text": "$QUERY"
                                            }
                                        }
                                    }
                                ]
                            }
                        },
                        "inner_hits": {
                            "_source": {
                                "excludes": [
                                    "passages.sparse"
                                ]
                            }
                        }
                    }
                }
            ],
            "filter": "$FILTER"
        }
    }
}
```

Is that right?  If so, I think it would be helpful for the document to list both and explain the difference.

Reply:
Good point. Will include both examples.